### ISSUE TITLE
word crashes when NVDA is active (on specific system)

### ISSUE THREAD

**Issue by FalkoBabbage** [nvaccess/nvda#14652]
### Steps to reproduce:
Open word and type something
### Actual behavior:

Word crashes
### Expected behavior:

Word does not crash
### NVDA logs, crash dumps and other attachments:
```
DEBUG - IAccessibleHandler.internalWinEventHandler.winEventCallback (10:13:38.900) - MainThread (18528):
Hook received winEvent: EVENT_OBJECT_NAMECHANGE, window 0 (unknown), objectID OBJID_CURSOR, childID 0, process 0 (unknown application), thread 18412
DEBUG - IAccessibleHandler.internalWinEventHandler.winEventCallback (10:13:38.901) - MainThread (18528):
Changing NULL or invalid window to desktop window for winEvent: EVENT_OBJECT_NAMECHANGE, window 0 (unknown), objectID OBJID_CURSOR, childID 0, process 0 (unknown application), thread 18412
DEBUG - IAccessibleHandler.internalWinEventHandler.winEventCallback (10:13:38.902) - MainThread (18528):
Adding winEvent to limiter: EVENT_OBJECT_NAMECHANGE, window 65552 (#32769), objectID OBJID_CURSOR, childID 0, process 948 (csrss), thread 18412
DEBUG - IAccessibleHandler.internalWinEventHandler.winEventCallback (10:13:38.964) - MainThread (18528):
Hook received winEvent: IA2_EVENT_TEXT_CARET_MOVED, window 459616 (_WwG), objectID 11, childID 0, process 8508 (winword), thread 18412
DEBUG - IAccessibleHandler.internalWinEventHandler.winEventCallback (10:13:38.965) - MainThread (18528):
Adding winEvent to limiter: IA2_EVENT_TEXT_CARET_MOVED, window 459616 (_WwG), objectID 11, childID 0, process 8508 (winword), thread 18412
DEBUGWARNING - RPC process 8508 (WINWORD.EXE) (10:13:39.348) - Dummy-8 (3304):
Thread 18412, build\x86_64\remote\winword.cpp, collectRevisionOffsets, 367:
range.revisions failed

IO - inputCore.InputManager.executeGesture (10:13:39.376) - winInputHook (5560):
Input: kb(desktop):i
IO - inputCore.InputManager.executeGesture (10:13:39.575) - winInputHook (5560):
Input: kb(desktop):k
IO - inputCore.InputManager.executeGesture (10:13:39.663) - winInputHook (5560):
Input: kb(desktop):space
DEBUGWARNING - watchdog._watcher (10:13:40.939) - watchdog (5112):
Trying to recover from freeze. Listing stacks for Python threads:
Python stack for thread 5560 (winInputHook):
  File "threading.pyc", line 890, in _bootstrap
  File "threading.pyc", line 926, in _bootstrap_inner
  File "threading.pyc", line 870, in run
  File "winInputHook.pyc", line 79, in hookThreadFunc

Python stack for thread 20316 (watchdog.CancellableCallThread.execute(<_FuncPtr object at 0x035FB990>)):
  File "threading.pyc", line 890, in _bootstrap
  File "threading.pyc", line 926, in _bootstrap_inner
  File "watchdog.pyc", line 332, in run
  File "threading.pyc", line 552, in wait
  File "threading.pyc", line 296, in wait

Python stack for thread 11728 (visionEnhancementProviders._base.wordFollower.rectHighlighter.rectHighlighter):
  File "threading.pyc", line 890, in _bootstrap
  File "threading.pyc", line 926, in _bootstrap_inner
  File "threading.pyc", line 870, in run
  File "C:\Users\XXXXX\AppData\Roaming\nvda\addons\USA\visionEnhancementProviders\_base\wordFollower\rectHighlighter.py", line 285, in _run
    while winUser.getMessage(byref(msg), None, 0, 0) > 0:
  File "winUser.pyc", line 427, in getMessage

Python stack for thread 5112 (watchdog):
  File "threading.pyc", line 890, in _bootstrap
  File "threading.pyc", line 926, in _bootstrap_inner
  File "threading.pyc", line 870, in run
  File "watchdog.pyc", line 126, in _watcher
  File "watchdog.pyc", line 62, in getFormattedStacksForAllThreads

Python stack for thread 4040 (touchHandler.TouchHandler):
  File "threading.pyc", line 890, in _bootstrap
  File "threading.pyc", line 926, in _bootstrap_inner
  File "touchHandler.pyc", line 238, in run

Python stack for thread 16348 (UIAHandler.UIAHandler.MTAThread):
  File "threading.pyc", line 890, in _bootstrap
  File "threading.pyc", line 926, in _bootstrap_inner
  File "threading.pyc", line 870, in run
  File "UIAHandler\__init__.pyc", line 371, in MTAThreadFunc
  File "queue.pyc", line 170, in get
  File "threading.pyc", line 296, in wait

Python stack for thread 12000 (braille._BgThread):
  File "threading.pyc", line 890, in _bootstrap
  File "threading.pyc", line 926, in _bootstrap_inner
  File "threading.pyc", line 870, in run
  File "braille.pyc", line 2294, in func

Python stack for thread 18528 (MainThread):
  File "nvda.pyw", line 389, in <module>
  File "core.pyc", line 776, in main
  File "wx\core.pyc", line 2237, in MainLoop
  File "gui\__init__.pyc", line 857, in Notify
  File "core.pyc", line 738, in run
  File "queueHandler.pyc", line 88, in pumpAll
  File "queueHandler.pyc", line 55, in flushQueue
  File "eventHandler.pyc", line 67, in _queueEventCallback
  File "eventHandler.pyc", line 284, in executeEvent
  File "eventHandler.pyc", line 98, in __init__
  File "eventHandler.pyc", line 107, in next
  File "NVDAObjects\__init__.pyc", line 1152, in event_gainFocus
  File "NVDAObjects\__init__.pyc", line 1052, in reportFocus
  File "speech\speech.pyc", line 558, in speakObject
  File "speech\speech.pyc", line 624, in getObjectSpeech
  File "speech\types.pyc", line 50, in _flattenNestedSequences
  File "speech\types.pyc", line 50, in <genexpr>
  File "speech\speech.pyc", line 1188, in getTextInfoSpeech
  File "NVDAObjects\window\winword.pyc", line 755, in getTextWithFields

DEBUGWARNING - NVDAObjects.window.winword.WordDocumentTextInfo.getTextWithFields (10:13:40.967) - MainThread (18528):
winword_getTextInRange failed with 1726
```
#### NVDA version:
2022.1
#### Windows version:
Not known
#### Name and version of other software in use when reproducing the issue:
Word 2208 Build 16.0.15601.20526 
#### Other information about your system:
This is on a computer 
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

I have seen this error on a computer of one of our customers. I cannot reproduce this error on my computer and her colleagues also do not have any trouble. I still would like to put this out since I am having trouble debugging and solving this problem. Word seems to crash after some RPC call made by NVDA. I have ran the COM registration fixing tool (and checked in the log if they were applied) and restarted the computer, but this did not seem to have any effect.
Are there any other routes I could take to see what goes wrong with this? The Event Viewer doesn't give any new information.

**Comment by Brian1Gaff** [comment]
Is the crash at all predictable? Does it also crash if you completely shut 
down nvda and use Narrator?
 First thoughts is that the installed windows and office are not the same as 
everyone else is using.

I'm very surprised that a registry fixing tool followed by a complete 
systemshutdowndoes not fix this.

Brian

-- 
***@***.***
Sent via blueyonder.(Virgin media)
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
----- Original Message ----- 
From: "FalkoBabbage" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Subscribed" ***@***.***>
Sent: Monday, February 20, 2023 10:11 AM
Subject: [nvaccess/nvda] word crashes when NVDA is active (on specific 
system) (Issue #14652)

**Comment by seanbudd** [comment]
2022.1 is an outdated version of NVDA, and this version of Word is outdated. Can you please update NVDA and Word.

**Comment by FalkoBabbage** [comment]
Hi, sorry for the late response.
The system where the crash occurs is maintained by an external party which makes it very difficult to just update MS Word. We did update NVDA to 2022.4 and the same problems occur.
I did recommend a re-install of windows itself because I think it might be a combination of windows updates, their "flavour" of windows and NVDA.
I was hoping if someone would have some tools to identify whether this is a problem with NVDA or with their windows / office installation. 
Since I do not have direct access to this system and I recommended a re-install anyway I will close this issue. If the issue persists after a full reinstall I will open this again.
